### PR TITLE
Fix/change delete file

### PR DESCRIPTION
**Pull request by nettashviro**
I changed the delete order between the file and its permissions, I also added a rollback mechanism that adds the permissions back in case something fails with the deleting of the file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The user in context is now more enriched (with fields like job, rank, current unit...)
 - The metrics metric now contains a timestamp and more information about the user.
 
+### Fixed
+- FIX([201](https://github.com/meateam/api-gateway/issues/201)): changed the delete order between the file and its permissions and added a rollback for the permissions
+
```
Comment (Shahar-Y) [inline CHANGELOG.md:19]: This should be under `Unreleased`

File: file/controller.go [hunk 2]
```diff
@@ -5,18 +5,55 @@ import (
...
+		return nil
+	}
+
+	// Delete file from db
+	deletedFile, err := fileClient.DeleteFileByID(ctx, &fpb.DeleteFileByIDRequest{Id: fileID})
+	if err != nil || deletedFile == nil {
+		loggermiddleware.LogError(logger, fmt.Errorf("failed deleting file: %v", err))
+
+		// Add permission rollback
+		AddPermissionsOnError(ctx, err, fileID, filePermissions.GetPermissions(), permissionClient, logger)
```
Comment (Shahar-Y) [inline file/controller.go:45]: What if the error is 404? Should we still restore the permissions? - The file does not exist

File: file/controller.go [hunk 3]
```diff
@@ -5,18 +5,55 @@ import (
...
 	fpb "github.com/meateam/file-service/proto/file"
 	ppb "github.com/meateam/permission-service/proto"
 	spb "github.com/meateam/search-service/proto"
 	upb "github.com/meateam/upload-service/proto"
 	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/status"
 )
 
+// deleteFileAndPrem deletes the file and the permissions to it from db
+func deleteFileAndPrem(ctx *gin.Context,
```
Comment (Shahar-Y) [inline file/controller.go:20]: Looks good, what about the permits of the file?

File: upload/upload.go [hunk 4]
```diff
@@ -920,7 +920,7 @@ func (r *Router) deleteOnError(c *gin.Context, err error, fileID string) {
 		return
 	}
 
-	_, deleteErr := file.DeleteFile(c.Request.Context(),
+	_, deleteErr := file.DeleteFile(c,
```
Comment (Shahar-Y) [inline upload/upload.go:923]: What is that?

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

**Comment by Shahar-Y** [comment]
Also, notice the branch is out-of-date

File: CHANGELOG.md (hunk 1 above)
Comment (nettashviro) [reply CHANGELOG.md:19]: done

File: file/controller.go (hunk 2 above)
Comment (nettashviro) [reply file/controller.go:45]: done

File: file/controller.go (hunk 3 above)
Comment (nettashviro) [reply file/controller.go:20]: In the original deletion, there is no reference to deleting a permit. I meant "perm" as a shortcut for "permissions".  I'll change the name of the function

File: upload/upload.go (hunk 4 above)
Comment (nettashviro) [reply upload/upload.go:923]: I changed DeleteFile function parameters from context.Context to gin.Context.
gin.Context already is a context.Context and implements all its functions. 
I had to use functionality that does not exist in context.Context (such as c.AbortWithStatus)

**Review by Shahar-Y — APPROVED** [review]
Don't push to develop

**Review by Shahar-Y — APPROVED** [review]

**Pull request merged**